ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlan_interfaces.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.convert_dicts import convert_dicts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.network_services.utils import UtilsMixin
+
+
+class VlanInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def vlan_interfaces(self):
+        """
+        Return structured config for vlan_interfaces
+
+        Consist of svis and mlag peering vlans from filtered tenants
+        """
+
+        if not (self._network_services_l2 and self._network_services_l3):
+            return None
+
+        vlans = {}
+        for tenant in self._filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                for svi in vrf["svis"]:
+                    if svi["nodes"]:
```

Comment:
I would split in method svi_vlan vs mlag_vlan to make the vlan_interfaces logic shorter

Reply:
done